src/motion_planning/src/plan_aak.cpp

```diff
@@ -0,0 +1,744 @@
+#include "ros/ros.h"
+#include <iomanip>
+#include <fstream>
+#include <sstream>
+#include <iostream>
+#include <stdlib.h>
+#include <bits/stdc++.h>
+#include <std_msgs/String.h>
+#include <cmath>
+#include <nav_msgs/OccupancyGrid.h>
+#include <nav_msgs/MapMetaData.h>
+#include <visualization_msgs/Marker.h>
+#include <nav_msgs/Path.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/Quaternion.h>
+#include <tf/transform_broadcaster.h>
+#include <tf/tf.h>
+#include <limits.h>
+
+//x_max = 4.65= -6.9+COLUMNS*0.05;  //y_max = 3.8= -5.9+ROWS*0.05;
+
+# define ROWS 194
+# define COLUMNS 231
+# define LENGTH ROWS*COLUMNS
+# define max numeric_limits<float>::infinity()
+
+using namespace std;
+
+
+typedef pair<int, int> pairs; 
+
+
+
+
+ float init_x = -6.9;
+ float init_y = -5.9;
+ float resolution = 0.05;
+
+
+
+
+class get_data {
+
+private:
+      
+      float heuristics(int current_node_row,int current_node_col,int goal_row,int goal_col);
+      void reshape(int x[LENGTH]);
+      auto get_grid_pos(float x, float y);
+
+public:
+
+ int counter = 0;
+ int x[LENGTH];
+ int total_path_cost = 1;
+ 
+
+ int start_row;
+ int start_col;
+ int goal_row; 
+ int goal_col; 
+ 
+ stack <pairs> path_taken ;
+  
+ void data();
+ void astar();
+ void get_destination(float x , float y);
+ void get_origin(float x , float y);
+ void path_traversed();
+
+ struct Node
+ {   
+ 	 float heur_cost;
+ 	 float move_cost ;
+ 	 float cost;
+ 	 int parent_node_row;
+ 	 int parent_node_col;
+     int cell_status ;
+     char cell_charc ;
+     bool visited ; 
+ } map[ROWS][COLUMNS];
+ 
+ };
+
+
+
+
+// function to get grid position corresponding to (x,y) coordinates
+
+auto get_data::get_grid_pos(float x,float y)
+{
+   pairs position;
+
+   int rows = (y-init_y)/resolution;     /// indicating Row 
```

Comment:
you are converting from a float to an int. if this is done deliberately then you should use typecasting to tell people that it is delibrate and not an oversight
```suggestion
   int rows = (int)((y-init_y)/resolution);     /// indicating Row 
```